lib/async_hooks.js

```diff
@@ -49,7 +49,12 @@ const before_symbol = Symbol('before');
 const after_symbol = Symbol('after');
 const destroy_symbol = Symbol('destroy');
 
-let setupHooksCalled = false;
+// Setup the callbacks that node::AsyncWrap will call when there are hooks to
+// process. They use the same functions as the JS embedder API.
+async_wrap.setupHooks({ init,
```

Comment:
Could you add a comment explaining why we do this immediately instead of waiting for the `AsyncHook.enable()` call.